scipy/io/mmio.py

```diff
@@ -317,8 +373,23 @@ def __init__(self, **kwargs):
         self._init_attrs(**kwargs)
 
     #---------------------------------------------------------------------------
-    def read(self, source):
-        stream, close_it = self._open(source)
+    def read(self, filename):
+        """
+        Reads the contents of a Matrix Market file 'filename' into a matrix.
+    
+        Parameters
+        ----------
+    
+        filename : file
```

Comment:
Rename to `file_like` or similar?  Type could be `file-like` too.  Presumably the object need not actually be a `file` but can be any file-like object (as well as a filename string).